packages/forms/src/UIForm/fields/Datalist/displayMode/TextMode.component.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { translate } from 'react-i18next';
+import callTrigger from '../Datalist.trigger';
+import { TextMode as FieldTemplate } from '../../FieldTemplate';
+import { I18N_DOMAIN_FORMS } from '../../../../constants';
+import theme from './TextMode.scss';
+import getDefaultT from '../../../../translate';
+
+class TextMode extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {};
+	}
+
+	componentDidMount() {
+		const onTrigger = (event, trigger) =>
+			this.props.onTrigger(event, {
+				trigger,
+				schema: this.props.schema,
+				errors: this.props.errors,
+				properties: this.props.properties,
+			});
+
+		callTrigger(null, {
+			eventNames: ['didMount', 'focus'],
+			triggersDefinitions: this.props.schema.triggers,
+			onTrigger,
+			onLoading: isLoading => this.setState({ isLoading }),
+			onResponse: data => this.setState(data),
+		});
+	}
+
+	render() {
+		const { id, schema, value, t } = this.props;
+		const { title } = schema;
+		const titleMap = this.state.titleMap || this.props.schema.titleMap || [];
+		const titleEntry = titleMap.find(entry => entry.value === value);
+
+		return (
+			<FieldTemplate id={id} label={title}>
+				{value && this.state.isLoading ? (
+					<span className={this.state.isLoading ? theme.loading : undefined} aria-busy="true">
+						{value} {`(${t('DATALIST_WIDGET_LOADING_LABELS', { defaultValue: 'loading labels' })})`}
+					</span>
+				) : (
+					(titleEntry && titleEntry.name) || value
+				)}
+			</FieldTemplate>
```

Comment:
Should be a simple `if/else`. It also looks like `this.state.isLoading` is tested twice.

Reply:
`It also looks like this.state.isLoading is tested twice.` --> thank you that survived my refacto :/